src/bridged_logging.rs

```diff
@@ -0,0 +1,226 @@
+// Copyright 2020, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+#![cfg(any(feature = "dox", all(feature = "v2_46", feature = "bridged_logging")))]
+
+extern crate log as rs_log;
+
+use glib_sys;
+use translate::*;
+
+/// Enumeration of the possible formatting behaviours for a [`GlibLogger`]
+#[derive(Debug, Copy, Clone)]
+pub enum GlibLoggerFormat {
+    /// A simple format, writing only the message on output.
+    Plain,
+    /// A simple format, writing file, line and message on output.
+    LineAndFile,
+    /// A logger using glib structured logging.
+    Structured,
+}
+
+/// Enumeration of the possible domain handling behaviours for a [`GlibLogger`]
+pub enum GlibLoggerDomain {
+    /// Logs will have no domain specified.
+    None,
+    /// Logs will use the `target` of the log crate as a domain; this allows
+    /// Rust code like `warn!(target: "my-domain", "...");` to log to the glib
+    /// logger using the specified domain.
+    CrateTarget,
+    /// Logs will use the crate path as the log domain.
+    CratePath,
+    /// Logs will use the specified domain for logging. Note that this option
+    /// due to the way the `log` crate is implemented, is global for all crates
+    /// in the current process.
+    Custom(&'static str),
```

Comment:
Is this really useful?

Reply:
I'm on the fence here.
On one side I assume there will be somebody who wants to add a Rust library to an existing glib based project and is happy to specify a single domain for all Rust logging; on the other they can get a better more future-proof result with a little more effort specifying a domain everywhere.

Reply:
I don't know, @GuillaumeGomez ? :)

Reply:
I think it's a good idea. However, a static `str` might be too restrictive.

Reply:
Must be either that or a `String`

Reply:
A `string` would be preferable indeed.

Reply:
I fear a `String` cannot be done (at the very least not without some dirty tricks), because the logger is a `&'static` and, as far as I can tell, there is no way to build a `static String` (all its constructors are not `const fn`).

Possible workarounds are with a `lazy_static` or with a `RefCell` inside the logger, but at that point I fear it becomes dirtier and harder to use correctly vs the potential gains.

*(disclaimer: limited knowledge of Rust here, so I might be missing something)

Reply:
You could use [`set_boxed_logger()`](https://docs.rs/log/0.4.11/log/fn.set_boxed_logger.html) but I agree that it's a bit ugly.

So what would be the use-case for this custom string, how would one use it and isn't it confusing to just have everything going into the `log` crate showing up the same way in the glib logs?

Reply:
My 2 cents given the current state: let's remove this option, move forward with the rest and eventually add it in the future in another pr if and when any reasonable need+solution comes up.

Reply:
Sounds good to me :)

Reply:
Sounds good to me as well.